src/rust/multi.nhentai/src/lib.rs

```diff
@@ -265,44 +213,46 @@ fn get_manga_details(id: String) -> Result<Manga> {
 
 #[get_chapter_list]
 fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
-	let request = Request::new(
-		helper::get_details_url(id.clone()).as_str(),
-		HttpMethod::Get,
-	)
-	.header("User-Agent", USER_AGENT);
-	let json = request.json()?.as_object()?;
+	let json = Request::new(helper::get_details_url(id.clone()), HttpMethod::Get)
+		.header("User-Agent", USER_AGENT)
+		.json()?
+		.as_object()?;
 
-	let mut url = String::from("https://nhentai.net/g/");
-	url.push_str(&id);
+	let url = format!("{BASE_URL}/g/{id}/");
 
 	let date_updated = json.get("upload_date").as_float().unwrap_or(0.0);
 
 	let language = &helper::get_tag_names_by_type(json.get("tags").as_array()?, "artist")?[0];
 
 	let lang = match language.as_str() {
-		"english" => String::from("en"),
-		"japanese" => String::from("jp"),
-		"chinese" => String::from("zh"),
-		_ => String::new(),
-	};
+		"english" => "en",
+		"japanese" => "jp",
+		"chinese" => "zh",
+		_ => "",
+	}
+	.to_string();
 
 	Ok(vec![Chapter {
 		id,
-		title: String::from("Chapter 1"),
-		volume: -1.0,
 		chapter: 1.0,
 		date_updated,
-		scanlator: String::new(),
 		url,
 		lang,
+		..Default::default()
 	}])
 }
 
 #[get_page_list]
 fn get_page_list(_: String, id: String) -> Result<Vec<Page>> {
-	let request = Request::new(helper::get_details_url(id).as_str(), HttpMethod::Get)
-		.header("User-Agent", USER_AGENT);
-	let json = request.json()?.as_object()?;
+	let html = Request::new(format!("{BASE_URL}/g/{id}/"), HttpMethod::Get)
+		.header("User-Agent", USER_AGENT)
+		.string()?;
```

Comment:
might be better to html this and maybe try and select the script tag, or around it, instead of loading the entire page into wasm memory, but probably fine